docs/migrating.adoc

```diff
@@ -0,0 +1,96 @@
+= Migrating your application between OpenShift environments
+
+:toc:
+:toc-placement!:
+
+This section describes the process of migrating your application between environments.
+
+'''
+toc::[]
+'''
+
+== When to migrate the application
+
+At some point you might need to transfer your application to a different environment.
+For example, in case you want to move from your development environment to a production
+environment, or if you want to make the application available to other developers.
+
+While you use ADB to develop your application, you define various configuration settings
+that support your application code. When you want to migrate your application, you export
+these settings to a configuration file and deploy it together with the code in the
+new environment.
+
+This configuration-driven approach allows you to manage each service of your application
+according to its configurations and share these configurations across environments.
+
+In OpenShift this type of migration is called Application Promotion. For general information about this process,
+see the https://docs.openshift.org/latest/dev_guide/application_lifecycle/promoting_applications.html[Promoting Applications Across Environments]
+
+
+== Exporting the application assets from the local environment
+
+In addition to migrating your application code, you need to export the configuration
+settings to a JSON file. These configuration settings represent metadata of the service,
+such as repository URL, route, build-config, networking, and so on.
+
+To export the configuration, run the following command for each container or
+service in your application:
+
+----
+oc export is,bc,dc,svc,route -o json > <service-name>.json
+----
+
+The following configuration objects are exported to the configuration file:
+
+//TODO: Add descriptions of each parameter
+
+* ImageStream (is)
+* BuildConfig (bc)
+* DeploymentConfig (dc)
+* Service (svc)
+* Route (route)
+
+//TODO: Need info about the other elements of the JSON file, example https://github.com/jbossdemocentral/coolstore-microservice/blob/master/openshift-templates/sso-service.json
+
```

Comment:
+1 @thatdocslady, liked the concept of `//TODO` for commenting in asciidoc, great for peer and Tech reviews and for getting inputs from SMEs. I was missing this from Google docs/Word.
Btw ran into this page while working on another issue: https://docs.openshift.org/latest/architecture/core_concepts/index.html maybe this could give more information about these elements/ maybe we can link to that doc. HTH.